suite/tests/linux/rseq.c

```diff
@@ -1123,9 +1127,15 @@ kernel_xfer_event(void *drcontext, const dr_kernel_xfer_info_t *info)
     assert(ok);
     /* All transfer cases in this test should have source info. */
     assert(info->source_mcontext != NULL);
-    if (info->type == DR_XFER_RSEQ_ABORT) {
-        /* The interrupted context should be identical except the pc. */
-        assert(info->source_mcontext->pc != mc.pc);
+    /* For a migration, the context should be identical.
+     * For a signal it won't be as signal handler arguments will be set in
+     * registers and a sigaltstack may be set.
+     */
+    if (info->type == DR_XFER_RSEQ_ABORT && in_rseq_migration_test) {
+        /* The interrupted context should be identical, including the pc now
+         * that DR passes the abort handler PC just like the kernel does.
+         */
+        assert(info->source_mcontext->pc == mc.pc);
 #    ifdef X86
         assert(info->source_mcontext->xax == mc.xax);
```

Comment:
Now that the pc is also the same, does the source_mcontext struct layout allow us to use a memcmp on x86 like we do on AArch64?

Reply:
Yes, can compare from xdi up through pc.  Done.